### PR TITLE
OP-1645: count start date of a product

### PR DESCRIPTION
**Pull request by olewandowski1**
[OP-1645](https://openimis.atlassian.net/browse/OP-1645)

[REQUIRED PR](https://github.com/openimis/openimis-fe-policy_js/pull/71)

Changes:
- Access to the product is restricted if the enrollment date precedes the product's 'dateFrom'.

[OP-1645]: https://openimis.atlassian.net/browse/OP-1645?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hooks.js
Comment (dragos-dobre) [inline src/hooks.js:30]: Why change this? Seems to work already.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: src/hooks.js
Comment (olewandowski1) [reply src/hooks.js:30]: This is necessary. I'll show you how it works before and after.
Before:
prev
Current: 
current
Product dateFrom is 2017-01-01. It means that it can not be used before this date. And without this change it is possible.

Comment (olewandowski1) [reply src/hooks.js:30]: It looks like it worked the other way around.

**Review by jdolkowski — APPROVED** [review]

**Comment by olewandowski1** [comment]
I've double checked it and it looks like it's working as expected. Please take a look on that:
Peek 2023-12-05 15-18
Validity of 2 products( BCUL0001 and FCTA0001 ) is from 2017-01-01 to 2030-12-31. And this validity is correctly evaluated. @delcroip

**Comment by olewandowski1** [comment]
Together with @wzgliniecki, we checked those dates and found nothing wrong. But the code is a bit confusing, though it works right now. Should we add some comments to it before merging, or should we hold off and make a ticket to look into it more? @delcroip @dragos-dobre

**Comment by delcroip** [comment]
We will merge with a comment

**Comment by olewandowski1** [comment]
@delcroip I've added comment, I'm moving PR from draft.

File: src/hooks.js
Comment (delcroip) [reply src/hooks.js:35]: Actually it is normal, those are just GQL params ...

**Pull request merged**
